lib/utilities/request.js

```diff
@@ -0,0 +1,161 @@
+const url = require('url');
+const http = require('http');
+const https = require('https');
+const HttpProxyAgent = require('http-proxy-agent');
+const HttpsProxyAgent = require('https-proxy-agent');
+
+const { jsutil } = require('arsenal');
+const {
+    proxyCompareUrl,
+} = require('arsenal').storage.data.external.backendUtils;
+
+const validVerbs = ['HEAD', 'GET', 'POST', 'PUT', 'DELETE'];
```

Comment:
I prefer Sets for checking contains as they are faster lookups, would prefer frozen sets or immutable sets but not sure how to do that in javascript.